op-e2e/actions/interop/dsl/interop.go

```diff
@@ -349,4 +374,83 @@ func (actors *InteropActors) ActBatchAndMine(t helpers.Testing, chains ...*Chain
 		actors.L1Miner.ActL1IncludeTxByHash(b.Hash())(t)
 	}
 	actors.L1Miner.ActL1EndBlock(t)
+
+	if cfg.shouldMarkSafe || cfg.shouldMarkFinal {
+		actors.L1Miner.ActL1SafeNext(t)
+	}
+	if cfg.shouldMarkFinal {
+		actors.L1Miner.ActEmptyBlock(t)
+		actors.L1Miner.ActL1FinalizeNext(t)
+	}
+}
+
+type actSyncSafeOption func(*actSyncSafeConfig)
+
+type actSyncSafeConfig struct {
+	ChainOpts
+	requireLocalSafeAdvances    bool
+	requireSafeAdvances         bool
+	requireFinalizedAdvances    bool
+	requireCrossUnsafeAdvances  bool
+	shouldSendL1FinalizedSignal bool
+	shouldSendL1LatestSignal    bool
+	isPreInterop                bool
```

Comment:
nit: I see that for every `isPreInterop` usage, it negates this value. If so, can we store the `isPostInterop` or `isInterop` and not negate for better readability?

Reply:
Removed in favor of assertions in {Superchain,Chain}SyncStatusAsserter